crates/solver/src/settlement.rs

```diff
@@ -140,14 +140,20 @@ impl Trade {
         self.compute_fee_execution(self.order.data.fee_amount)
     }
 
-    /// Returns the scaled unsubsidized fee amount that should be used for
-    /// objective value computation.
-    pub fn executed_scaled_unsubsidized_fee(&self) -> Option<U256> {
-        self.compute_fee_execution(self.scaled_unsubsidized_fee)
+    /// Returns the solver fee used for computing the objective value adjusted
+    /// for partial fills.
+    pub fn executed_solver_fee(&self) -> Option<U256> {
+        self.compute_fee_execution(self.solver_fee)
     }
 
-    pub fn executed_unscaled_subsidized_fee(&self) -> Option<U256> {
-        self.compute_fee_execution(self.order.data.fee_amount)
+    /// Returns the actual fees taken by the protocol.
+    pub fn executed_earned_fee(&self) -> Option<U256> {
+        let surplus_fee = match &self.order.metadata.class {
+            // TODO adjust this when limit orders become partially fillable.
+            OrderClass::Limit(details) => details.surplus_fee.unwrap_or_default(),
+            _ => 0.into(),
+        };
```

Comment:
This was not respected before. But it doesn't matter much, since this function was only used in debug log statements.

Reply:
Nice catch!